claims.go

```diff
@@ -1,177 +1,16 @@
 package jwt
 
-import (
-	"crypto/subtle"
-	"fmt"
-	"time"
-)
-
-// Claims must just have a Valid method that determines
-// if the token is invalid for any supported reason
+// Claims represent any form of a JWT Claims Set according to
```

Comment:
Have you considered an alternate scenario where we remove this (and map claims) entirely?

It might be a bad idea, but just throwing this thought out .. the user still has to assert the claims.

Reply:
I think a lot of people still use map claims, so I am not 100 % convinced tbh.

Reply:
Ye that's fair. My thought was .. there are too many ways to pass "claims"

1. map claims (default)
2. RegisteredClaims (structured, arguably could be the default?)
3. Custom claims that either implement Claims or embed RegisteredClaims

If we remove `Claims` interface entirely, and take an `any` .. would that simply the overall API?

Probably a bad idea, figured I'd bring it up but I think you're probably right .. a lot of people use map claims.

Reply:
I think it actually gets easier with the new interface, because you do not need to implement any validation on your claim, but rather you can use any kind of struct/map type that supplies the needed standard claim values, such as exp, iat, etc. I think that is a pretty straight forward way of modelling claims now.